### PR TITLE
Add `black-primer` docs

### PR DESCRIPTION
**Pull request by cooperlees**
- Document the idea, CLI args, config and a example run for `black-primer` in README.md
- Add to docs/index.rst

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1117,6 +1118,124 @@ Apart from the above, `blackd` can produce the following response codes:
 The response headers include a `X-Black-Version` header containing the version of
 _Black_.
 
+## black-primer
+
+`black-primer` is a tool built for CI (and huumans) to have black `--check` a number of
```
Comment (JelleZijlstra) [inline README.md:1123]:
```suggestion
`black-primer` is a tool built for CI (and humans) to have black `--check` a number of
```

File: README.md [hunk 2]
```diff
@@ -1117,6 +1118,124 @@ Apart from the above, `blackd` can produce the following response codes:
...
+### Run flow
+
+- Ensure we have a `black` + `git` in PATH
+- Load projects from `primer.json`
+- Run projects in parallel with `--worker` workers (defaults to CPU Count / 2)
+  - Checkout projects
+  - Run black and record result
+  - Clean up repository checkout _(can optionally be disabled via `--keep`)_
+- Display results summary to screen
+- Default to cleaning up `--work-dir` (which defaults to tempfile schemantics)
```
Comment (JelleZijlstra) [inline README.md:1136]:
```suggestion
- Default to cleaning up `--work-dir` (which defaults to tempfile semantics)
```

File: README.md [hunk 3]
```diff
@@ -1117,6 +1118,124 @@ Apart from the above, `blackd` can produce the following response codes:
...
+                         ders/tc/hbwxh76j1hn6gqjd2n2sjn4j9k1glp/T/primer.20200
+                         517125229]
+
+  -W, --workers INTEGER  Number of parallel worker coroutines  [default: 69]
+  -h, --help             Show this message and exit.
+```
+
+### primer config file
+
+The config is `JSON` format. It's main element is the `"projects"` dictionary. Below
```
Comment (JelleZijlstra) [inline README.md:1176]:
```suggestion
The config is `JSON` format. Its main element is the `"projects"` dictionary. Below
```

**Comment by cooperlees** [comment]
Thanks - I am the typo king :(

**Comment by zsol** [comment]
Content lgtm. I was wondering if we should break up the massive readme and move these additional tools (along with `blackd`) into separate doc files linked to from the main one. Wdyt?

**Comment by cooperlees** [comment]
Sure, I would prefer that too. But I feel that’s a different PR where no content changes and also think we should design it together to avoid PR churn?

**Comment by zsol** [comment]
Sounds good!

**Pull request merged**